### PR TITLE
Add types via JSDoc

### LINKED ISSUE #2: Unit test defect (does not impact package functionality)
I'm not a remark expert but looking at the parsed result in the debugger...  

I believe the unit test: 'should find with string condition', should be passing 'text' as the second parameter, not 'value'.

### LINKED ISSUE #7: Ts type?
Seems this util package didn't generate dts file like other packages.

### PR DESCRIPTION
**Pull request by paescuj**
Closes #7, closes #4 as replacement (and also closes #2 since I made sure the test is valid)

Additionally, this pull request updates the dependencies.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -4,19 +4,34 @@
...
 import iteratee from 'lodash.iteratee'
 
 /**
- * Find
+ * Unist node finder utility.
  *
- * @param {Node} tree - Root node
- * @param {string|object|function} [condition] - Condition to match node.
+ * @param tree
+ *   Node to search
```
Comment (paescuj) [inline index.js]:
```suggestion
 *   Node to search.
```

**Comment by joshuadelange** [comment]
This PR would really help us a lot. Is there any possibility to get this merged & released?

If it needs more work we can consider helping out a bit too.

cc: @blahah

**Comment by paescuj** [comment]
Ready from my side! Happy to do some final adjustments if necessary.

**Comment by blahah** [comment]
Sorry for the delay - looks good, merging

**Comment by paescuj** [comment]
> Sorry for the delay - looks good, merging

Awesome, thanks! 🎉 No worries, that's how things are in the world of open source 😃

**Pull request merged**